InvenTree/plugin/base/barcodes/mixins.py

```diff
@@ -54,60 +53,218 @@ def scan(self, barcode_data):
+                "error": _("Found multiple matching supplier parts for barcode")
+            }
+
+        if len(supplier_parts) == 0:
+            return None
+
+        # Extract the first supplier part
+        supplier_part = supplier_parts[0]
+
+        if self.quantity is not None and self.quantity <= 0:
+            return {
+                "error": _("Quantity must be greater than zero")
+            }
+
+        # Find matching purchase order (if not supplied)
+        if not purchase_order:
+            matching_orders = self.get_purchase_orders(
+                purchase_order=purchase_order,
+                customer_order_number=self.customer_order_number,
+                supplier_order_number=self.supplier_order_number,
+                supplier=self.get_supplier(),
+            )
+
+            if len(matching_orders) == 1:
+                purchase_order = matching_orders[0]
+
+            elif len(matching_orders) > 1:
+                return {
+                    "error": _("Found multiple matching purchase orders for barcode")
+                }
+
+            elif len(matching_orders) == 0:
+                return {
+                    "error": _("No matching purchase orders found for barcode")
+                }
+
+        return self.receive_purchase_order_item(
+            supplier_part,
+            purchase_order,
+            user,
+            quantity=self.quantity,
+            location=location,
+            barcode=barcode_data,
+        )
+
+    @property
+    def SKU(self):
+        """Return the SKU field data."""
+        return self.get_field_data(self.SUPPLIER_PART_NUMBER)
+
+    @property
+    def MPN(self):
+        """Return the MPN field data"""
+        return self.get_field_data(self.MANUFACTURER_PART_NUMBER)
+
+    @property
+    def quantity(self):
+        """Return the quantity field data."""
+        q = self.get_field_data(self.QUANTITY)
+
+        if q is not None:
+            try:
+                return Decimal(q)
+            except InvalidOperation:
+                pass
+
+    @property
+    def customer_order_number(self):
+        """Return the customer (internal) order number"""
+        return self.get_field_data(self.CUSTOMER_ORDER_NUMBER)
+
+    @property
+    def supplier_order_number(self):
+        """Return the supplier's order number"""
+        return self.get_field_data(self.SUPPLIER_ORDER_NUMBER)
+
+    def get_field_data(self, code):
+        """Return the field data for a given field code."""
+
+        fields = getattr(self, 'barcode_fields') or []
+
+        # Return the first "match" for the given code
+        for field in fields:
+            if field.startswith(code):
+                return field[len(code):]
+
         return None
 
-    @staticmethod
-    def parse_ecia_barcode2d(barcode_data: str | list[str]) -> dict[str, str]:
-        """Parse a standard ECIA 2D barcode, according to https://www.ecianow.org/assets/docs/ECIA_Specifications.pdf"""
```

Comment:
Comment: This function is IMO very helpful, I would leave it in. The function returned a lot more info than the rewrite and is very usefull to extract more info from barcodes.